src/containers/addNew.js

```diff
@@ -11,140 +11,51 @@ const emptyColors = 'rgb(255,239,212)';
-
-  handleChange = field => event => {
-    this.setState(
-      {
-        [field]: event.target.value,
-      },
-      () => {
-        this.setButtonDisabledValue(this.state.beerName, this.state.desc);
-      },
-    );
-  };
-
-  validate = field => () => {
+  setFileName = fileName => {
     this.setState({
-      touched: { ...this.state.touched, [field]: true },
+      file: fileName,
     });
   };
 
   renderField = (label, field) => {
-    let classnames = null;
-
-    if (field === 'beerName')
-      classnames = classNames('form-group', {
-        'has-danger': this.state.touched.beerName && this.state.beerName === '',
-      });
-    else
-      classnames = classNames('form-group', {
-        'has-danger': this.state.touched.desc && this.state.desc === '',
-      });
+    let classnames = 'form-group';
 
     return (
       <div className={classnames}>
         <label className="field-label">{label}</label>
-        <input
-          className="form-control"
-          type="text"
-          value={field === 'beerName' ? this.state.beerName : this.state.desc}
-          onChange={this.handleChange(field)}
-          onBlur={this.validate(field)}
-        />
+        <Field className="form-control" type="text" name={field} />
       </div>
     );
   };
 
-  renderNumberField = field => {
-    return (
-      <div className="form-group" id="rating-field">
-        <label className="field-label">Enter the rating</label>
-        <label className="stars">
-          <Ratings
-            rating={this.state.rating}
-            widgetEmptyColors={emptyColors}
-            widgetHoverColors={hoverColors}
-            changeRating={this.changeTheRating}
-            widgetDimensions={dimensions}
-          >
-            <Ratings.Widget />
-            <Ratings.Widget />
-            <Ratings.Widget />
-            <Ratings.Widget />
-            <Ratings.Widget />
-          </Ratings>
-        </label>
-        <input
-          type="number"
-          className="form-control rating-input"
-          onChange={this.handleChange(field)}
-          value={this.state.rating}
-          readOnly
-        />
-      </div>
-    );
-  };
-
-  renderImgUpload = () => {
-    return (
-      <div className="file-area">
-        <label className="field-label">Image: &nbsp; </label>
-        <div className="file-overlay">
-          <input
-            className="file-input"
-            type="file"
-            accept=".png, .jpeg, .jpg"
-          />
-        </div>
-      </div>
-    );
-  };
-
-  onSubmit = e => {
-    e.preventDefault();
-
+  onSubmit = values => {
     let imgName = null,
       file = null;
 
-    //check if there is a filedata object and filedata object is not null
-    if (e.target[3].files && e.target[3].files[0]) {
-      imgName = e.target[3].files[0].name;
-      file = e.target[3].files[0];
+    if (values.file) {
+      imgName = values.file.name;
+      file = values.file;
     }
+    values.img = imgName;
```

Comment:
You are modifying props data, which should be immutable/readonly.
Formik has its api to update values